### DISCUSSION TITLE
How to add a payload for the server sent events?

### DISCUSSION THREAD

**Discussion by manzaloros** [sergiodxa/remix-utils#396] [Q&A]
How can I send an object payload to the `eventStream` function in my resource route from the UI?

Let's say I want to send an object like
```
{
  foo: 'bar',
  baz: [
    {
       qux: 'hello world'
    }
  ]
}
```

to start the eventStream. The object is dependent on some form data from the user on the UI.

How would I do this?

**Comment by sergiodxa** [comment]
Convert the value to a string, then parse it client-side
```ts
// server-side
let data = {
  foo: 'bar',
  baz: [
    {
       qux: 'hello world'
    }
  ]
}
send({ data: JSON.stringify(data) })
```

```ts
// client-side
let data = useEventSource("/endpoint");
let parsed = data ? JSON.parse(data) : null)
// use parsed
```

**Reply by manzaloros** [reply]
Thanks! The object I'm sending might be too large for that (isn't there a limit to the string size here?), so I'm thinking your example of using [EventEmitters](https://sergiodxa.com/tutorials/use-server-sent-events-with-remix) might be the way to go. 

1. Submit an object as a POST request to a route.
2. Get a transaction ID back from the API server and emit an event
3. Use the transaction ID returned from the fetcher in `useEventSource` as the message ID
4. in `useEventSource` call some streaming API with that message.id to receive the streaming response from the server

How does that sound?
